library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -1436,6 +1475,11 @@ impl<T, A: Allocator> VecDeque<T, A> {
     /// deque.clear();
     /// assert!(deque.is_empty());
     /// ```
+    /// # Complexity
+    ///
+    /// Takes `O(n)` time using [`truncate`] to drop elements.
```

Comment:
This is also potentially misleading. For Vec there's been a lot of discussion (I'm not sure on the current state) about whether we should guarantee O(1) truncate/clear for elements that don't need drop (e.g., u8) even in debug mode.